lib/util/user_account_channel.dart

```diff
@@ -2,50 +2,119 @@ import 'dart:async';
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/service/configuration_service.dart';
 import 'package:autonomy_flutter/util/log.dart';
 import 'package:autonomy_flutter/util/wallet_utils.dart';
 import 'package:flutter/services.dart';
 
-class PrimaryAddressChannel {
+class UserAccountChannel {
   final MethodChannel _channel;
 
-  PrimaryAddressChannel()
+  UserAccountChannel()
       : _channel = Platform.isIOS
             ? const MethodChannel('migration_util')
             : const MethodChannel('backup');
 
+  String? _userId;
+  AddressInfo? _primaryAddress;
+
   Future<void> setPrimaryAddress(AddressInfo info) async {
     try {
       await _channel
           .invokeMethod('setPrimaryAddress', {'data': info.toString()});
+      _primaryAddress = info;
     } catch (e) {
       log.info('setPrimaryAddress error: $e');
     }
   }
 
   Future<AddressInfo?> getPrimaryAddress() async {
+    if (_primaryAddress != null) {
+      return _primaryAddress;
+    }
     try {
       final String data = await _channel.invokeMethod('getPrimaryAddress', {});
       if (data.isEmpty) {
-        return null;
+        _primaryAddress = null;
       }
       final primaryAddressInfo = json.decode(data);
```

Comment:
should add else when data.isNotEmpty

Reply:
shouldn't check at all, fromJson will fail anyway